### ISSUE TITLE
 VideoCapture buffer

### ISSUE THREAD

**Issue by mgobishop** [justadudewhohacks/opencv4nodejs#382]
I am using cv.VideoCapture but my video has come from a web service and is in a buffer (e.g. as base64). Do I have to make a temp file so that I can pass a path to VideoCapture or is there some other neater way? I’m using cv.imdecode for images with no need for a temp file.

**Comment by mgobishop** [comment]
It’s a buffered video i.e. not a file so no path to pass to VideoCapture. I need to loop over the frames of the video hence I need to use VideoCapture. Any ideas? Maybe I’m going about it the wrong way?

**Comment by mgobishop** [comment]
Any ideas?

**Comment by justadudewhohacks** [comment]
Not sure what your "buffered video" actually looks like, but as I said, if it's a buffer of base64 encoded frames, then my guess is to pull frame by frame from the buffer and decode them using cv.imdecode.

**Comment by mgobishop** [comment]
Ok, so maybe my question is how do I pull the frames from the video. I am using OpenCV's VideoCapture but I have to make a temp file first. I hoped my code could look like:

    server.js
    app.post('/api/doc/upload', (req, res, next) => {
	    Analyser.classifyVideo(req.body.data);
    }

    Analyser.js
    static async classifyVideo(buffer) {
        let results = [];
        const cap = new cv.VideoCapture(HAS TO BE A FILE PATH!!!); // I CANT USE MY BUFFER
        this.loadNeuralNet();
        let f = 0;
        return new Promise((resolve, reject) => {
            let ti = setInterval(() => {
                f = f + 1;
                let frame = cap.read();
                if (frame.empty == true) {
                    clearInterval(ti);
                    resolve({ frames: f, predictions: results });
                }
                if (f % 10 == 0) results = results.concat(this.onFrame(frame, f));
                cv.waitKey(1);
              }, 0);
          })
      }

Thanks for any advice...

**Comment by SystemDiagnosticss** [comment]
@mgobishop are you solved?? Also I need capture video from a web service and I can't send it in socket.io without saving ...

**Comment by mgobishop** [comment]
Nope! All I can think of doing is saving the temp file to RAM Disk - but I don't like it!
